Client/Frontend/Sync/SyncPairCameraViewController.swift

```diff
@@ -181,11 +142,112 @@ class SyncPairCameraViewController: SyncViewController {
         }
     }
     
-    @objc func SEL_enterWords() {
+    @objc
+    private func onEnterWordsPressed() {
         let wordsVC = SyncPairWordsViewController(syncAPI: syncAPI)
-        wordsVC.syncHandler = self.syncHandler
+        wordsVC.delegate = delegate
         navigationController?.pushViewController(wordsVC, animated: true)
     }
+    
+    private func onQRCodeScanned(data: String) {
+        // Guard against multi-scanning
+        if cameraLocked { return }
+        cameraLocked = true
+        
+        // Pause scanning
+        cameraView.cameraOverlaySuccess()
+        cameraView.stopRunning()
+        
+        // Vibrate.
+        AudioServicesPlayAlertSound(SystemSoundID(kSystemSoundID_Vibrate))
+        
+        // Forced timeout
+        let task = DispatchWorkItem { [weak self] in
+            guard let self = self else { return }
+            self.cameraLocked = false
+            self.cameraView.cameraOverlayError()
+        }
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 25.0, execute: task)
```

Comment:
Can we add move this 25 expiration to a static value in this class? So it might easier to track later.